scipy/io/mmio.py

```diff
@@ -505,7 +532,7 @@ def _parse_body(self, stream):
         return a
 
     #---------------------------------------------------------------------------
-    def _write(self, stream, a, comment='', field=None, precision=None):
+    def _write(self, stream, a, comment='', field=None, precision=None, symmetry=None):
```

Comment:
Just to confirm - is this right - that the previous default was to check symmetry for array-like input, but not for sparse input?

If so, that is an odd default.  Unless you can think of a good reason to continue with that default (I can't), then it seems  more sensible to go for `symmetry=None` default for all matrices.   The user can turn always off the check by passing something explicit for symmetry.